pallets/collection-data-feed/src/lib.rs

```diff
@@ -152,9 +159,11 @@ pub mod pallet {
 			// for Data values.
 			for collection_id in Listening::<T>::get(data_id).keys() {
 				Collection::<T>::mutate(collection_id, |collection| {
-					collection
-						.get_mut(data_id)
-						.map(|value| *value = Self::get(data_id))
+					if let Some(value) = collection.get_mut(data_id) {
```

Comment:
I'm wondering if It might read a little cleaner flattened with match, though feel free to disregard if not :)
```
match (collection.get_mut(data_id), Self::get(data_id)) {
    (Some(value), Ok(new_value)) => *value = new_value,
    _ => ()
}
```